app/controllers/members_controller.rb

```diff
@@ -17,9 +18,13 @@ def edit
   end
 
   def update
+    if params['member']['static_slots_attributes'].present?
+      Member::StaticSlotAssigner.call(current_member, permitted_params['static_slots_attributes'])
+    end
+
```

Comment:
The User should see a flash message if his/her static slot assignement has failed for whatever reason.

Since there are now two distinct steps in this `#update` action, and you would want to set errors feedback messages according to a given step, you might like to use [this gem](https://dry-rb.org/gems/dry-transaction/0.13/) to manage this through 'Railway programming' style.

So you could create a 'Transaction' object, to call from this action. The result of the transaction would then guide the action.
In the 'Transaction' object, you'll have 2 steps: `assign_static_slots` and `update_attributes`. If one step fails, this step returns a `Failure` with a custom Hash containing whatever info you need (e.g. `return Failure({errors: 'this is a custom error message})`)
The controller action would behave according to that Transaction's result

e.g.:
```ruby
def update
  transaction = Members::UpdateTransaction.call(params: permitted_params)
  if transaction.success?
      flash[:notice] = t 'activerecord.notices.messages.update_success'
      redirect_to member_path(@member.id) 
  else
      flash[:error] = transaction[:errors]
      render :edit 
  end
end
```

Using this, the controller stays simple, and the feedback messages setup logic is moved to the Transaction object